File: Leet-Code/Python/Satcks/min-stack.py
class MinStack(object):

    def __init__(self):
        self.data = []
        self.head = -1
        self.len = 0
        

    def push(self, val):
        if self.len == 0:
            self.head = 0
        
        self.data.append(val)
        self.head += 1
        self.len += 1

    def pop(self):
        if self.len == 0:
            return 
        self.data.pop()
        self.len -= 1
        self.head -= 1
    

    def top(self):
        return self.data[self.len - 1]

    def getMin(self):
        m = 10000000
        for i in range(len(self.data)):
            if m > self.data[i]:
                m = self.data[i]
       
        print(m) 

def main():
    l1 = MinStack()
    l1.push(-2)
    l1.push(0)
    l1.push(-3)
    l1.getMin() 
    l1.pop()
    print(l1.top())
    l1.getMin()
    
    
    
main()

File: Leet-Code/Stacks/min_stack.cc
#include <climits>
#include <cstring>
#include <iostream>
#include <vector>

class MinStack {
public:
  std::vector<int> stack;
  MinStack(){};

  void push(int val) { stack.push_back(val); }

  void pop() { stack.pop_back(); }

  int top() { return stack.back(); }

  int getMin() {
    int min = INT_MAX;
    for (int n : stack) {
      if (n < min)
        min = n;
    }
    return min;
  };
};

/**
 * Your MinStack object will be instantiated and called as such:
 * MinStack* obj = new MinStack();
 * obj->push(val);
 * obj->pop();
 * int param_3 = obj->top();
 * int param_4 = obj->getMin();
 */
int main(int argc, char *argv[]) {

  MinStack x;

  return 0;
}
